src/dynarec/la64/dynarec_la64_00.c

```diff
@@ -476,26 +472,8 @@ uintptr_t dynarec64_00(dynarec_la64_t* dyn, uintptr_t addr, uintptr_t ip, int ni
                     eb1 = TO_LA64((nextop & 7) + (rex.b << 3));
                     eb2 = 0;
                 }
-                if (eb2) {
-                    // load a mask to x3 (ffffffffffff00ff)
-                    LU12I_W(x3, 0xffff0);
-                    ORI(x3, x3, 0xff);
-                    // apply mask
-                    AND(eb1, eb1, x3);
-                    if (u8) {
-                        if ((u8 << 8) < 2048) {
-                            ADDI_D(x4, xZR, u8 << 8);
-                        } else {
-                            ADDI_D(x4, xZR, u8);
-                            SLLI_D(x4, x4, 8);
-                        }
-                        OR(eb1, eb1, x4);
-                    }
-                } else {
-                    ADDI_W(x3, xZR, 0xf00); // mask ffffffffffffff00
-                    AND(eb1, eb1, x3);
-                    ORI(eb1, eb1, u8);
-                }
+                MOV32w(x3, u8);
+                BSTRINS_D(eb1, x3, eb2 * 8 + 7, 8);
```

Comment:
shouldn't this be
```C
 BSTRINS_D(eb1, x3, eb2 * 8 + 7, eb2 * 8);
```
instead?

Reply:
Yeah, thank you for saving the future me.